types/ir/ir_test.go

```diff
@@ -0,0 +1,287 @@
+		service := NewServiceWithName(serviceName)
+		service.AddPortForwarding(servicePort, podPort, serviceRelPath)
+		found := false
+		for i := range service.ServiceToPodPortForwardings {
+			if reflect.DeepEqual(service.ServiceToPodPortForwardings[i], serviceToPodPortForwarding) {
+				found = true
+			}
+		}
+		if !found {
+			t.Error("failed to add port forwarding to service")
+		}
+	})
+
+	t.Run("test for adding exposed port to container", func(t *testing.T) {
+		container := NewContainer()
+		var exposedPort int32 = 8080
+		container.AddExposedPort(exposedPort)
+		found := false
+
+		for i := range container.ExposedPorts {
+			if container.ExposedPorts[i] == exposedPort {
+				found = true
+			}
+		}
+		if !found {
+			t.Error("failed to add exposed port to container")
+		}
+	})
+
+	t.Run("test for adding accessed dirs to container", func(t *testing.T) {
+		container := NewContainer()
+		var accessedDir string = "dir1"
+		container.AddAccessedDirs(accessedDir)
+		found := false
+
+		for i := range container.AccessedDirs {
+			if container.AccessedDirs[i] == accessedDir {
+				found = true
+			}
+		}
+		if !found {
+			t.Error("failed to add accessed dir to container")
+		}
+	})
+
+	t.Run("test for having valid annotation", func(t *testing.T) {
+		service := NewServiceWithName("sampleService")
+		service.Annotations = map[string]string{common.WindowsAnnotation: common.AnnotationLabelValue}
+		if !service.HasValidAnnotation(common.WindowsAnnotation) {
+			t.Error("failed to validate annotations of the service")
+		}
+	})
+
+	t.Run("test for getting all service ports", func(t *testing.T) {
+		ir := NewIR()
+		var port int32 = 8080
+		service := Service{Name: "sampleService", ServiceToPodPortForwardings: []ServiceToPodPortForwarding{{PodPort: networking.ServiceBackendPort{Name: "somePort", Number: port}}}}
+		ir.AddService(service)
+		ports := ir.GetAllServicePorts()
+		found := false
+		for i := range ports {
+			if ports[i] == port {
+				found = true
+			}
+		}
+		if !found {
+			t.Error("failed to get service ports")
+		}
+	})
+
+	t.Run("test for merging storages", func(t *testing.T) {
+		oldStorage := Storage{Name: "sampleStorage", StorageType: "pvc"}
+		newStorage := Storage{Name: "sampleStorage", StorageType: "cfmap"}
+		oldStorage.Merge(newStorage)
+		if oldStorage.Name != newStorage.Name || oldStorage.StorageType != newStorage.StorageType {
+			t.Errorf("failed to merge new storage into old storage. Old Storage : %+v, New Storage: %+v", oldStorage, newStorage)
+		}
+	})
+
+	t.Run("test for merging IRs", func(t *testing.T) {
+		oldIR := NewIR()
+		newIR := NewIR()
+		oldIR.Name = "sampleIR"
+		newIR.Name = "sampleIR"
+		newIR.AddService(NewServiceWithName("sampleSerivce"))
+		oldIR.Merge(newIR)
+		if !reflect.DeepEqual(oldIR, newIR) {
+			t.Errorf("failed to merge new IR into old IR. Old IR : %+v, New IR: %+v", oldIR, newIR)
+		}
+	})
+
+	t.Run("test for merging containers", func(t *testing.T) {
+		oldContainer := NewContainer()
```

Comment:
Let's add more comprehensive tests instead of using empty objects.
Most of the merge functionality will be skipped if the fields in the object are empty/nil.

We can `logrus.Infof("%#v", ir)` in the `ir.go` file and run a proper `move2kube transform -s language-platforms` to get a nearly full IR. Will be closer to real world use.

Reply:
I've addressed this in the new commits. Please review, Thanks!